### PR TITLE
[tests-only][full-ci] Test for opening password protected folder with wrong password

### PR DESCRIPTION
**Pull request by anon-pradip**
## Description
This PR contains test steps to check if password protected folder can't be opened with wrong password and proper error message appears for trying to do so in both project and personal space

## Related Issue

Related to: https://github.com/owncloud/web/issues/12181

## How Has This Been Tested?

- test environment:
- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature [hunk 1]
```diff
@@ -34,6 +34,15 @@ Feature: password-protected folder operations
     And "Alice" copies the link of password protected folder "folder1.psec"
     And "Alice" closes the password protected folder modal
 
+    # Trying to open with wrong password
+    And "Alice" opens folder "folder1.psec"
+    When "Alice" tries to unlock password protected folder with password "wrong-password"
+    Then "Alice" should see an error message in the password protected folder modal
+      """
+      Incorrect password
+      """
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature:43]: did we had a bug for this?

File: tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature [hunk 2]
```diff
@@ -34,6 +34,15 @@ Feature: password-protected folder operations
...
+    And "Alice" closes the password protected folder modal
```
Comment (PrajwolAmatya) [inline tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature:44]:
```suggestion
    And "Alice" opens folder "folder1.psec"
    When "Alice" tries to unlock password protected folder with password "wrong-password"
    Then "Alice" should see an error message in the password protected folder modal
      """
      Incorrect password
      """
    And "Alice" unlocks password protected folder with password "%public%"
    And "Alice" copies the link of password protected folder "folder1.psec"
    And "Alice" closes the password protected folder modal
```

Can we do this order just adding wrong password step after line 32?

File: tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature [hunk 3]
```diff
@@ -34,6 +34,15 @@ Feature: password-protected folder operations
...
+
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature:45]: I would prefer this flow:
```feature
When "Alice" tries to unlock password protected folder with password "wrong-password"
And "Alice" unlocks password protected folder with password "%public%"
And "Alice" copies the link of password protected folder "folder1.psec"
And "Alice" closes the password protected folder modal
```

Comment (saw-jan) [reply tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature:45]: check error within `tries to...` step

File: tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature [hunk 4]
```diff
@@ -116,6 +134,15 @@ Feature: password-protected folder operations
...
 
+    # Trying to open with wrong password
+    And "Brian" opens folder "space-folder1.psec"
+    When "Brian" tries to unlock password protected folder with password "wrong-password"
+    Then "Brian" should see an error message in the password protected folder modal
+      """
+      Incorrect password
+      """
+    And "Brian" closes the password protected folder modal
+
```
Comment (PrajwolAmatya) [inline tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature:145]: IMO the above check by user `Alice` would be sufficient.

File: tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature (hunk 1 above)
Comment (anon-pradip) [reply tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature:43]: This is for edge case mentioned here: https://github.com/owncloud/web/issues/12181

Comment (saw-jan) [reply tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature:43]: we should be careful about adding edge cases and negative webUI tests.

Comment (saw-jan) [reply tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature:43]: we usually don't add negative webUI tests

Comment (saw-jan) [reply tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature:43]: if there were no bugs related to it then I would not add a test for it.

Comment (saw-jan) [reply tests/e2e/cucumber/features/file-action/passwordProtectedFolder.feature:43]: for this PR, let's add this steps only in this scenario

File: tests/e2e/support/store/link.ts [hunk 5]
```diff
@@ -10,3 +10,4 @@ export const roleDisplayText: Record<string, string> = {
   'Secret File Drop': 'Secret File drop'
 }
 export const securePassword = 'Pwd:12345'
+export const incorrectPasswordMessage = 'Incorrect password'
```
Comment (saw-jan) [inline tests/e2e/support/store/link.ts:13]: I didn't know that we have this step:
```
Then "Alice" should see an error message
  """
  message
  """
```
you can utilize it if it exists.

File: tests/e2e/cucumber/steps/ui/public.ts [hunk 6]
```diff
@@ -217,3 +217,20 @@ When(
...
+    const pageObject = new objects.applicationFiles.page.Public({ page })
+    const linkObject = new objects.applicationFiles.Link({ page })
+    password = password === '%public%' ? securePassword : password
+    await pageObject.authenticate({
+      password,
+      passwordProtectedFolder: true,
+      expectToSucceed: false
+    })
+    const actualErrorMessage = await linkObject.checkErrorMessage({ passwordProtectedFolder: true })
+    expect(actualErrorMessage).toBe(incorrectPasswordMessage)
```
Comment (saw-jan) [inline tests/e2e/cucumber/steps/ui/public.ts:234]:
```suggestion
    expect(actualErrorMessage).toBe("Incorrect password")
```

File: tests/e2e/support/store/link.ts (hunk 5 above)
Comment (saw-jan) [inline tests/e2e/support/store/link.ts:13]: this is most likely one time use. so no need to save it
```suggestion
```

**Review by saw-jan — APPROVED** [review]

**Review by amrita-shrestha — APPROVED** [review]
